### PR TITLE
take domain into account when checking for wiki resources

### PR DESCRIPTION
**Pull request by joewalsh**
https://phabricator.wikimedia.org/T131078

### CODE REVIEW

**Comment by montehurd** [comment]
Article links stopped working on today's featured article. iPhone 6s 9.3

**Comment by joewalsh** [comment]
@montehurd added fix

File: Wikipedia/Code/NSURL+WMFLinkParsing.m [hunk 1]
```diff
@@ -180,7 +180,13 @@ + (NSURL *)wmf_desktopURLForURL:(NSURL *)url {
 #pragma mark - Properties
 
 - (BOOL)wmf_isWikiResource {
-    return [self.path wmf_isWikiResource];
```
Comment (montehurd) [inline Wikipedia/Code/NSURL+WMFLinkParsing.m]: That last commit fixed the article link taps. 

Was wondering if it's ok that this logic now diverges from the NSString `wmf_isWikiResource` that it had formerly called? Should that method be updated or should this logic move there?

Comment (joewalsh) [inline Wikipedia/Code/NSURL+WMFLinkParsing.m]: wmf_isWikiResource on a string should only be used for paths. If anything, we can remove that public interface and only allow checks on a URL

Comment (coreyfloyd) [inline Wikipedia/Code/NSURL+WMFLinkParsing.m]: Sounds good to me… prefer to get rid of duplicate APIs

File: Wikipedia/Code/NSURL+WMFLinkParsing.m [hunk 2]
```diff
@@ -180,7 +180,13 @@ + (NSURL *)wmf_desktopURLForURL:(NSURL *)url {
...
+    static NSString *wikiResourceSuffix = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        wikiResourceSuffix = [NSString stringWithFormat:@".%@", WMFDefaultSiteDomain];
+    });
+    NSString *lowercaseHost = self.host.lowercaseString;
+    return (!lowercaseHost || [lowercaseHost isEqualToString:WMFDefaultSiteDomain] || [lowercaseHost hasSuffix:wikiResourceSuffix]) && [self.path wmf_isWikiResource];
```
Comment (coreyfloyd) [inline Wikipedia/Code/NSURL+WMFLinkParsing.m]: `[lowercaseHost isEqualToString:WMFDefaultSiteDomain]` - that doesn't mean a wiki resource I don't think

Comment (coreyfloyd) [inline Wikipedia/Code/NSURL+WMFLinkParsing.m]: Or wait… never mind…

If we check for  "wikipedia.org", we also need to check for other "known" domains. For instance this breaks for "mediawiki.org"

Comment (coreyfloyd) [inline Wikipedia/Code/NSURL+WMFLinkParsing.m]: Just for reference… our FAQ is on mediawiki.org and follows the same logic. There was a poorly named `wmf_isNonStandardURL` property to check URLs like this that don't have language domains and special case some things.

**Pull request merged**